scripts/ci/collect-splunk-logs.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# Collect logs from as splunk pod daemon
+#
+# Usage:
+#   collect-splunk-logs.sh NAMESPACE DIR
+
+usage() {
+    echo "./scripts/ci/collect-splunk-logs.sh <namespace> <output-dir>"
+}
+
+main() {
+    if [[ "$#" -lt 2 ]]; then
+        usage
+        exit 1
+    fi
+
+    namespace="$1"
+    log_dir="$2"
+
+    if ! kubectl get ns "${namespace}"; then
+        echo "Collection is skipped when the splunk namespace is missing: ${namespace}"
+        exit 0
+    fi
+
+    mkdir -p "${log_dir}"
+
+    splunk_pod="$(kubectl -n "${namespace}" get pods -o json \
+      | jq -r '.items[] | select(.metadata.labels["app"]? | match("^splunk")) | .metadata.name')"
+
+    kubectl -n "${namespace}" exec "${splunk_pod}" -- \
```

Comment:
Can you remind me - doesn't Splunk show the same logs as the pod logs?

Reply:
@ivan-degtiarenko TL;DR: The pod logs of Splunk do not contain any operational output.

Why?
The Splunk Docker image is built on Ansible. 
If you run the image, a full Ansible run takes place to deploy Splunk inside the container.
This means that the pod logs only show the Ansible logs, which finish the moment Splunk starts 😄 
All relevant logs that we are interested in, e.g. the ACS TA logs, or the splunkd logs, are located in `/opt/splunk/var/log/splunk`. 
As the Splunk integration test has been flaky and we don't really know why, it might be worth the storage to collect detailed logs of the TA and/or the Splunk deployment to narrow down what is wrong with the deployment when the integration test failes.